src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -434,7 +434,16 @@ bool DirectoryService::RunConsensusOnDSBlockWhenDSPrimary()
     // Add the oldest DS committee member to m_allPoWs and m_allPoWConns so it gets included in sharding structure
     sortedPoWSolns.emplace_back(array<unsigned char, 32>(),
                                 m_mediator.m_DSCommittee->back().first);
-    m_allPoWConns.emplace(m_mediator.m_DSCommittee->back());
+
+    if (m_mediator.m_DSCommittee->back().first == m_mediator.m_selfKey.second)
+    {
+        m_allPoWConns.emplace(make_pair(m_mediator.m_DSCommittee->back().first,
```

Comment:
This implictly link  ```pubkey``` of ```m_DSCommitteee->back()``` to the node's ip address. While this is true, more explicit pairing is preferred. 

```m_allPoWConns.emplace(make_pair(m_mediator.m_selfKey.second,
                                        m_mediator.m_selfPeer));```

Reply:
noted